src/CartesianProduct.jl

```diff
@@ -549,3 +551,68 @@ function vertices_list(cpa::CartesianProductArray{N}
 
     return vlist
 end
+
+"""
+
+    block_indices(ca::CartesianProductArray{N}, vars::Vector{Int}) where {N}
+
+Return the dictionary of non-universal blocks (a key is the index of the block and a value is the index of the first variable in the block according to the whole set) of a (polytopic) Cartesian product of a finite
+number of sets.
+
+### Input
+
+- `ca` -- Cartesian product array
+- `vars` -- List of non free variables
+
+### Output
+
+A dictionary of block indices and indices of their first variables in the whole system.
+
+"""
+function block_indices(ca::CartesianProductArray{N}, vars::Vector{Int}) where {N}
+    #key is the index of block, value is the index of the starting variable
+    result = Dict{Int,Int}()
+    start_index = 1
+    if isempty(vars)
+        for i in 1:length(ca.array)
+            result[i] = start_index
+            start_index += dim(ca.array[i])
+        end
+    else
+        for var in vars
```

Comment:
i think that this logic is.. wrong? let the block partition be `[1,2], [3], [4,5,6]`. and suppose that `vars = [4]`. 

- at the beginning `start_index_1` and the loop in `i` is from `1` to `3`. 
- first `start_index <= var < start_index_dim(..) => 1 <= 4 < 3 => false`; 
- then `1 <= 4 < 1+1=2 => false`, and `1 <= 4 < 4 => false`.

so the `result` is empty. but i think that it should be the dictionary with keys `[3]` and values `[4]`?

probably the `start_index` variable has to be updated if the check in parentheses fails.